### PR TITLE
[FIX] Use username when display name empty

### LINKED ISSUE #1260: Users with no display name set should display username in profile
- Your Rocket.Chat Experimental app version: **Testflight** 1.20.0 (11407)
- Your Rocket.Chat server version: 1.2.1
- Device (or Simulator) you're running with: iPhone 8

See attached screenshot for a user without any display name set. It would be nice if this still displayed something, even Unnamed (+ username below) like the web desktop version.

image1

### PR DESCRIPTION
**Pull request by aashaypalliwar**
@RocketChat/ReactNative

Closes #1260

### CODE REVIEW

File: app/views/RoomActionsView/index.js [hunk 1]
```diff
@@ -417,7 +417,7 @@ class RoomActionsView extends React.Component {
 				</Avatar>,
 				<View key='name' style={styles.roomTitleContainer}>
 					{room.t === 'd'
-						? <Text style={[styles.roomTitle, { color: themes[theme].titleText }]} numberOfLines={1}>{room.fname}</Text>
+						? <Text style={[styles.roomTitle, { color: themes[theme].titleText }]} numberOfLines={1}>{room.fname || `${ name }` || 'unnamed'}</Text>
```
Comment (djorkaeffalexandre) [inline app/views/RoomActionsView/index.js]: Add I18n translations to `unnamed`.
Use => `room.fname || name || I18n.t('unnamed')`

File: app/views/RoomInfoView/index.js [hunk 2]
```diff
@@ -285,6 +285,8 @@ class RoomInfoView extends React.Component {
 	render() {
 		const { room, roomUser } = this.state;
 		const { theme } = this.props;
+		let displayName = roomUser.name || roomUser.username;
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: Create a function:
getName = (roomUser) => {
  if (roomUser) {
    return roomUser.name || roomUser.username || I18n.t('unnamed');
  }
  return I18n.t('unnamed');
}

File: app/views/RoomInfoView/index.js [hunk 3]
```diff
@@ -298,7 +300,7 @@ class RoomInfoView extends React.Component {
 				>
 					<View style={styles.avatarContainer}>
 						{this.renderAvatar(room, roomUser)}
-						<View style={styles.roomTitleContainer}>{ getRoomTitle(room, this.t, roomUser && roomUser.name, theme) }</View>
+						<View style={styles.roomTitleContainer}>{ getRoomTitle(room, this.t, roomUser && displayName, theme) }</View>
```
Comment (djorkaeffalexandre) [inline app/views/RoomInfoView/index.js]: If you use the above comment you can use only `getName(roomUser)` here.

**Comment by diegolmello** [comment]
@aashaypalliwar Please, stop updating your branch for every new commit that lands.
Update it only if there's a conflict.
We're going to review your PR soon.
Thanks.

File: app/views/RoomActionsView/index.js [hunk 4]
```diff
@@ -417,7 +417,7 @@ class RoomActionsView extends React.Component {
 				</Avatar>,
 				<View key='name' style={styles.roomTitleContainer}>
 					{room.t === 'd'
-						? <Text style={[styles.roomTitle, { color: themes[theme].titleText }]} numberOfLines={1}>{room.fname}</Text>
+						? <Text style={[styles.roomTitle, { color: themes[theme].titleText }]} numberOfLines={1}>{room.fname || name || I18n.t('unnamed')}</Text>
```
Comment (djorkaeffalexandre) [inline app/views/RoomActionsView/index.js]: Add `unnamed` translation to English and your native language.. :)

**Pull request closed without merging**